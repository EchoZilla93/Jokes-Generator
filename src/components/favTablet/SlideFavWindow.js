import React from 'react';
import FavoriteCardBluePrint from '../Favorite/FavoriteCardBluePrint';


const SlideFavWindow = () => {
    return(
        <div className="favoritesList">
            <div className="favoritesList__item">
                <FavoriteCardBluePrint/>
            </div>
        </div>
    )
}

export default SlideFavWindow;