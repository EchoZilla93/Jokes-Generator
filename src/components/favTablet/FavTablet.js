import React,{useState} from 'react';

import './favTablet.scss';
import SlideFavWindow from './SlideFavWindow';

const FavTablet = () => {
    const [favClick,setFavClick] = useState(false);

    const windowToogle = () => {
        setFavClick(true);
            if(favClick === true){
                return setFavClick(false)
            }
    }
    return(
        <div className="favoriteTabletView">
            <button onClick={windowToogle} className="favoriteTabletView__openFav">
                <i className={favClick?"fas fa-times" : "fas fa-bars"}></i>
            </button>
            <h1 className="favoriteTabletView__heading">Favorite</h1>
            {favClick && <SlideFavWindow/>}
        </div> 
    )
};

export default FavTablet;