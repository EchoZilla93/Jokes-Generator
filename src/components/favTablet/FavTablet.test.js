import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import FavTablet from './FavTablet';
import SlideFavWindow from  './SlideFavWindow';

describe('favorite section components', () => {
    it('renders expect snapshot', () => {
        const wrapper = shallow(<SlideFavWindow/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    it('renders expect snapshot',() => {
        const wrapper = shallow(<FavTablet/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})


