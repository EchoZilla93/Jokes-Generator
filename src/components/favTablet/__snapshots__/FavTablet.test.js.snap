// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favorite section components renders expect snapshot 1`] = `
<div
  className="favoritesList"
>
  <div
    className="favoritesList__item"
  >
    <FavoriteCardBluePrint />
  </div>
</div>
`;

exports[`favorite section components renders expect snapshot 2`] = `
<div
  className="favoriteTabletView"
>
  <button
    className="favoriteTabletView__openFav"
    onClick={[Function]}
  >
    <i
      className="fas fa-bars"
    />
  </button>
  <h1
    className="favoriteTabletView__heading"
  >
    Favorite
  </h1>
</div>
`;
